contrib/src/main/java/com/datatorrent/contrib/parser/CsvParser.java

```
@@ -62,247 +71,193 @@
+    processors = getProcessor(delimitedParserSchema.getFields());
+    csvStringReader = new ReusableStringReader();
+    csvMapReader = new CsvMapReader(csvStringReader, preference);
+    csvBeanReader = new CsvBeanReader(csvStringReader, preference);
   }
 
-  private void initialise(String[] nameMapping, CellProcessor[] processors)
-  {
-    for (int i = 0; i < getFields().size(); i++) {
-      FIELD_TYPE type = getFields().get(i).type;
-      nameMapping[i] = getFields().get(i).name;
-      if (type == FIELD_TYPE.DOUBLE) {
-        processors[i] = new Optional(new ParseDouble());
-      } else if (type == FIELD_TYPE.INTEGER) {
-        processors[i] = new Optional(new ParseInt());
-      } else if (type == FIELD_TYPE.FLOAT) {
-        processors[i] = new Optional(new ParseDouble());
-      } else if (type == FIELD_TYPE.LONG) {
-        processors[i] = new Optional(new ParseLong());
-      } else if (type == FIELD_TYPE.SHORT) {
-        processors[i] = new Optional(new ParseInt());
-      } else if (type == FIELD_TYPE.STRING) {
-        processors[i] = new Optional();
-      } else if (type == FIELD_TYPE.CHARACTER) {
-        processors[i] = new Optional(new ParseChar());
-      } else if (type == FIELD_TYPE.BOOLEAN) {
-        processors[i] = new Optional(new ParseBool());
-      } else if (type == FIELD_TYPE.DATE) {
-        String dateFormat = getFields().get(i).format;
-        processors[i] = new Optional(new ParseDate(dateFormat == null ? "dd/MM/yyyy" : dateFormat));
-      }
-    }
-  }
   @Override
-  public Object convert(String tuple)
+  public Object convert(byte[] tuple)
   {
-    try {
-      csvStringReader.open(tuple);
-      return csvReader.read(clazz, nameMapping, processors);
-    } catch (IOException e) {
-      logger.debug("Error while converting tuple {} {}",tuple,e.getMessage());
-      return null;
-    }
+    //This method is not invoked for CSV parser
+    return null;
```

If this is not invoked, please throw UnsupportedOperationException rather then returning any value.